riot/runtime/scheduler/scheduler.ml

```diff
@@ -333,6 +335,20 @@ module Scheduler = struct
 
   let tick_timers _pool (sch : t) = Timer_wheel.tick sch.timers
 
+  let steal_processes pool sch =
```

Comment:
This looks great! Super straight forward. Makes me think that `currently_stealing` could actually be a mutex, so we can wrap this up in a `Mutex.protect` call:

```
Mutex.protect @@ fun () ->
  List.iter (fun scheduler ->
    if Mutex.try_lock scheduler.currently_stealing then
       (* do stealing *)
       Mutex.unlock scheduler.currently_stealing 
  ) pool.schedulers)
```

Something like this, so this guarantees that if A is stealing from B, then A won't get stolen from C, and noone else would be able to steal from B either.

Reply:
That makes sense!  With this, we are guaranteed that a process will only be stolen by one scheduler a time, so it is OK to leave the `sid` in `Process.t` as mutable rather than an `Atomic.t`, right?  (No need to respond if this is correct.)